vsphere/datadog_checks/vsphere/api.py

```diff
@@ -91,6 +91,8 @@ def smart_connect(self):
             err_msg = "Connection to {} failed: {}".format(self.config.hostname, e)
             raise APIConnectionError(err_msg)
 
+        if self._conn:
+            connect.Disconnect(self._conn)
```

Comment:
Note for reviewer, this call cannot fail:
- https://github.com/vmware/pyvmomi/blob/97d91fe194778e9f6da131d50402e3706708a9e5/pyVim/connect.py#L310-L317
- https://github.com/vmware/pyvmomi/blob/97d91fe194778e9f6da131d50402e3706708a9e5/pyVim/connect.py#L453-L463